pkg/requesthandler/accounts.go

```diff
@@ -99,7 +99,7 @@ func (r *RequestHandler) ValidatorByAccountAddress(accountAddress *iotago.Accoun
 func (r *RequestHandler) RewardsByOutputID(outputID iotago.OutputID, slot iotago.SlotIndex) (*api.ManaRewardsResponse, error) {
 	utxoOutput, err := r.protocol.Engines.Main.Get().Ledger.Output(outputID)
 	if err != nil {
-		return nil, ierrors.Wrapf(echo.ErrInternalServerError, "failed to get output %s from ledger: %s", outputID.ToHex(), err)
+		return nil, ierrors.Wrapf(echo.ErrNotFound, "output %s not found in the Ledger: %s", outputID.ToHex(), err)
```

Comment:
This would return `ErrNotFound` in every possible error case.
It should only be `ErrNotFound` if the database entry was not found, otherwise it should be `ErrInternalServerError`.